docs/rfcs/0000-namespace.md

```diff
@@ -0,0 +1,76 @@
+- Start Date: (2015-10-29)
+- RFC PR: (leave this empty)
+- AMS Issue: (leave this empty)
+
+# Summary
+
+Provide a consistent API for the user of the AMS.
+
+# Motivation
+
+The actual public API is defined under `ActiveModelSerializers`,
+`ActiveModel::Serializer` and `ActiveModel`.
```

Comment:
reposting https://github.com/rails-api/active_model_serializers/issues/1298#issuecomment-151267763 here

So, I haven't thought it all the way through, and I just wrote it out without any review, but what I'm thinking right now is something like:

| Current | Desired | Notes |
|----|---|------|
| require 'active_model_serializers' | require 'active_model_serializers'
| ActiveModelSerializers | ActiveModelSerializers 
| ActiveModel::SerializableResource | ActiveModel::SerializableResource
| namespace ActiveModel::Serializer | ActiveModelSerializers::Current, ActiveModelSerializers::V08, etc | versioned
| class ActiveModel::Serializer  | ActiveModelSerializers::Current::Resource | is a class that defines the properties and behavior of the data that you present to the user
|  ActiveModelSerializers::Model |  ActiveModelSerializers::Model | e.g. ActiveRecord object but a PORO. Maybe should be ActiveModelSerializers::Record A record is an instance of a model that contains data loaded from a server. Your application can also create new records and save them back to the server.
| ActiveModel::Serializer::Lint | ActiveModelSerializers::Model::Lint
| ActiveModel::Serialization | ActiveModelSerializers::ActionController::Serialization 
| ActiveModel::Serializer::CollectionSerializer | ActiveModelSerializers::Current::ResourceCollection |
| namespace ActiveModel::Serializer::Adapter | ActiveModelSerializers::Adapter |   adapter: knows how to talk to your server. knows how to translate requests from AMS into requests on your server.  object that translates requests from AMS/Rails/Grape etc (such as "find the user with an ID of 123") into a requests to a server.  let you completely change how your API is implemented without impacting your application code.
| class ActiveModel::Serializer::Adapter::JsonApi | ActiveModelSerializers::Adapter::JsonApi, ActiveModelSerializers::Adapter::V08::Json | not versioned
| class ActiveModel::Serializer, ActiveModel::Serializer::Adapter::Attributes | ActiveModelSerializers::Formatter | maps keys and values to desired format, see https://github.com/orbitjs/orbit.js/blob/master/lib/orbit-common/jsonapi-serializer.js
| N/A |  ActiveModelSerializers::Current::Store | store is the central repository of records in your application, use the store to retrieve records, as well to create new ones. The store will automatically cache records for you. 
| N/A | JsonApi::Exceptions 
| ActiveModel::Serializer::Adapter<br>::JsonApi::ApiObject::JsonApi | ActiveModelSerializers::Adapter::JsonApi::ApiObject::{JsonApi,Links,Link,Resource,Error, etc}? vs. JsonApiFormat https://github.com/endpoints/endpoints/blob/master/es5/format-jsonapi/index.js or even schema https://github.com/orbitjs/orbit.js/blob/master/lib/orbit-common/schema.js https://github.com/endpoints/endpoints/blob/master/es5/validate-json-schema/index.js
| ActiveModelSerializers.logger | ActiveModelSerializers.logger 

see 
- http://guides.emberjs.com/v2.1.0/models/
- https://github.com/cerebris/jsonapi-resources/blob/master/lib/jsonapi/formatter.rb
- https://andycrum.github.io/ember-data-model-maker/
- https://github.com/cerebris/jsonapi-resources#jsonapiresource
- origins of naming in AMS:
  - https://github.com/rails/rails/commit/d2b78b3594b9cc9870e6a6ebfeb2e56d00e6ddb8#diff-80d5beeced9bdc24ca2b04a201543bdd
  - https://github.com/rails/rails/commit/783db25e0c640c1588732967a87d65c10fddc08e
  - https://github.com/rails/rails/commit/c6bc8e662614be711f45a8d4b231d5f993b024a7
  - https://github.com/rails/rails/commit/fbdf706fffbfb17731a1f459203d242414ef5086#diff-531b52ff1432c7c9f9472188b39d6a43
  - https://github.com/rails/rails/commit/4bfbdc133a37c6046b8854d3659ba75597d1d37e
  - https://github.com/rails/rails/commit/8896b4fdc8a543157cdf4dfc378607ebf6c10ab0
  - https://github.com/rails-api/active_model_serializers/commits/master?page=19
  - https://github.com/rails-api/active_model_serializers/commit/729a823868e8c7ac86c653fcc7100ee511e08cb6#diff-fe7aa2941c19a41ccea6e52940d84016
  - https://github.com/rails-api/active_model_serializers/commit/4a2d9853ba7486acc1747752982aa5650e7fd6e9
  - https://github.com/rails-api/active_model_serializers/commit/f00fe5595ddf741dc26127ed8fe81adad833ead5